json-rpc/src/runtime.rs

```diff
@@ -116,10 +122,20 @@ async fn rpc_request_handler(
 ) -> Value {
     let request: Map<String, Value>;
     let mut response = Map::new();
+    let version = ledger_info.ledger_info().version();
+    let timestamp = ledger_info.ledger_info().timestamp_usecs();
 
     // set defaults: protocol version to 2.0, request id to null
     response.insert("jsonrpc".to_string(), Value::String("2.0".to_string()));
     response.insert("id".to_string(), Value::Null);
+    response.insert(
+        JSONRPC_LIBRA_LEDGER_VERSION.to_string(),
+        Value::Number(version.into()),
+    );
+    response.insert(
+        JSONRPC_LIBRA_LEDGER_TIMESTAMPUSECS.to_string(),
+        Value::Number(timestamp.into()),
+    );
```

Comment:
2 concerns:
1. We found the version and timestamp before execute, this is confusing as response, because I thought it is version and timestamp after method is executed.
2. this does not match jsonrpc spec standard, which means, it's possible a standard jsonrpc client may not be able to parse these data out, if it only returns result / error from response.

A nit concern is get_metadata also response these 2 values, the response will be odd.

Reply:
resolved offline